name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/handlers/UploadToMyDividends24Handler.java

```diff
@@ -147,3 +172,147 @@ protected IStatus run(IProgressMonitor monitor)
         }.schedule();
     }
 }
+
+class PortfolioSelectionDialog24 extends Dialog
+{
+
+    private List<Object> portfolios;
+    private List<Pair<Integer, String>> myDividendsPortfolios;
+
+    private Object selectedPortfolio;
+    private Pair<Integer, String> selectedMyDividendsPortfolio;
+
+    private TableViewer portfoliosTableViewer;
+    private TableViewer myDividendsTableViewer;
+
+    private boolean includeTransactions = false;
+
+    protected PortfolioSelectionDialog24(Shell parentShell)
+    {
+        super(parentShell);
+    }
+
+    public void setMyDividendsPortfolio(List<Pair<Integer, String>> myDividendsPortfolios)
+    {
+        this.myDividendsPortfolios = myDividendsPortfolios;
+    }
+
+    public void setPortfolio(List<Object> portfolios)
+    {
+        this.portfolios = portfolios;
+    }
+
+    public Object getSelectedPPPortfolio()
+    {
+        return selectedPortfolio;
+    }
+
+    public Pair<Integer, String> getSelectedMyDividendsPortfolio()
```

Comment:
Naming: make it consistent to whatever you choose above (see the first comment about MY24Portfolio).